cmd/pipeline/config.go

```diff
@@ -181,4 +181,8 @@ func configure(v *viper.Viper, p *pflag.FlagSet) {
 
 	v.SetDefault("spotmetrics::enabled", false)
 	v.SetDefault("spotmetrics::collectionInterval", 30*time.Second)
+
+	// Cadence configuration
+	v.SetDefault("cadence::createNonexistentDomain", false)
+	v.SetDefault("cadence::workflowExecutionRetentionPeriodInDays", 3)
```

Comment:
Why is this added here? AFAIK the worker is responsible for creating the domain.

Reply:
It's in the PR description, but additional info is that it was introduced in #3319 because without this I was unable to bootstrap a test where the worker starts independently from the pipeline process, and the pipeline process always started sooner. If for some reason (concurrency?) we don't want this, then I think I can add this to the test setup only.

Reply:
I don't really like the idea TBH. The fact that Pipeline (worker) creates the domain for itself is bad to begin with, but now the API will try that, even in production. But this is probably the easier solution for now. If you can avoid it in the future, great, but let's not block this one for now.